src/main/java/squid/Squid.java

```diff
@@ -0,0 +1,353 @@
+            throw new NotEnoughInputsException(
+                    String.format(Exceptions.NOT_ENOUGH_INPUTS,
+                            "delete",
+                            CorrectUsage.DELETE));
+        }
+        Task deleted = Tasks.delete(params[1]);
+        echo(String.format(Messages.DELETE, params[1], deleted));
+        Tasks.list();
+    }
+
+
+
+    private static void find(String input) throws NotEnoughInputsException {
+        String[] params = input.split(" ", 2);
+        if (params.length == 1) {
+            throw new NotEnoughInputsException(
+                    String.format(Exceptions.NOT_ENOUGH_INPUTS, "find", CorrectUsage.FIND));
+        }
+        String keyword = params[1];
+        echo(Messages.FIND);
+        Tasks.find(keyword);
+    }
+
+    /**
+     * Parse the user's input and assigns them to separate helper functions depending on command
+     *
+     * @param loop Condition whether to terminate loop.
+     * @param input The user's input.
+     * @return Whether the loop should continue (Usually true unless "bye" command is given).
+     * @throws SquidException General exception thrown by Squid.
+     */
+    private static boolean parseInput(boolean loop, String input) throws SquidException {
+        System.out.println(Messages.LINE_BREAK);
+        String[] messages = input.split(" ", 2);
+        String command = messages[0];
+        String params = "";
+        if (messages.length > 1) {
+            params = messages[1];
+        }
+
+        try {
+            switch (command) {
+            case ("bye"):
+                loop = false;
+                bye();
+                break;
+            case ("echo"):
+                echo(input, true);
+                break;
+            case ("list"):
+                list();
+                break;
+            case ("mark"):
+                mark(input, true);
+                break;
+            case ("unmark"):
+                mark(input, false);
+                break;
+            case ("todo"):
+                todo(input);
+                break;
+            case ("deadline"):
+                deadline(input);
+                break;
+            case ("event"):
+                event(input);
+                break;
+            case ("delete"):
+                delete(input);
+                break;
+            case ("save"):
+                Tasks.save();
+                echo(Messages.SAVE);
+                break;
+            case ("find"):
+                find(input);
+                break;
+            default:
+                throw new IncorrectInputException(Exceptions.INCORRECT_INPUT);
+            }
+        } catch (SquidException e) {
+            echo(e.getMessage());
+        }
+        return loop;
+    }
+
+
+
+    public static void main(String[] args) throws SquidException {
+        new Squid();
+        hello();
+
+        Scanner scanner = new Scanner(System.in);
+        boolean loop = true;
```

Comment:
Would it be better to name the loop variable to sound like a boolean (eg. isLoop)?